controllers/clusteroperator.go

```diff
@@ -0,0 +1,229 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	osconfigv1 "github.com/openshift/api/config/v1"
+	osclientset "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// StatusReason is a MixedCaps string representing the reason for a
+// status condition change.
+type StatusReason string
+
+const (
+	clusterOperatorName = "baremetal"
+
+	// CO Status Condition Types
+	Available   osconfigv1.ClusterStatusConditionType = osconfigv1.OperatorAvailable
```

Comment:
Can we just use the existing constants from the osconfigv1 package instead of defining and exporting local versions of them here?

Reply:
Sure. I think it made it easier to read but I'm indifferent. Will revert.